modules/@apostrophecms/asset/lib/webpack/media-to-container-queries-loader.js

```diff
@@ -0,0 +1,71 @@
+module.exports = function (source) {
+  const schema = {
+    title: 'Media to Container Queries Loader options',
+    type: 'object',
+    properties: {
+      debug: {
+        type: 'boolean'
+      },
+      transform: {
+        anyOf: [
+          { type: 'null' },
+          { instanceof: 'Function' }
+        ]
+      }
+    }
+  };
+  const options = this.getOptions(schema);
+
+  const mediaQueryRegex = /@media\s*(all|print|screen(?: and | or )?)?([^{]+)[^{]*{([\s\S]*?})\s*(\\n)*}/g;
+
+  const convertToContainerQuery = (mediaFeature, content) => {
+    // NOTE: container query does not work with the combo
+    // - min-width, max-width, min-height, max-height
+    // - lower than, lower than equal, greater than, greater than equal
+    const DESCRIPTORS = [
+      'min-width',
+      'max-width',
+      'min-height',
+      'max-height'
+    ];
+    const OPERATORS = [
+      '>',
+      '>=',
+      '<',
+      '<='
+    ];
+
+    const containerFeature = typeof options.transform === 'function'
+      ? options.transform(mediaFeature)
+      : mediaFeature;
+
+    if (
+      options.debug &&
+      DESCRIPTORS.some(descriptor => containerFeature.includes(descriptor)) &&
+      OPERATORS.some(operator => containerFeature.includes(operator))
+    ) {
+      console.warn('[mediaToContainerQueryLoader] Unsupported media query', containerFeature);
+    }
+
+    return `@container ${containerFeature} {${content}}`;
+  };
+
+  // Prepend container query to media queries
+  const modifiedSource = source.replace(mediaQueryRegex, (match, mediaType, mediaFeature, content) => {
+    if (
+      mediaType === 'print' ||
+      (
+        mediaType !== undefined &&
+        ([ 'all', 'screen' ].some(media => mediaType.includes(media))) === false
+      )
+    ) {
+      return match;
+    }
+
+    const containerQuery = convertToContainerQuery(mediaFeature, content);
+
+    return `${containerQuery} ${match}`;
```

Comment:
```suggestion
    return `${containerQuery}\n\n${match}`;
```

(approximative code)

to place the media query on a new line

Reply:
I've also put the container query after the media query